src/constants.ts

```diff
@@ -1,5 +1,127 @@
+import path from 'path';
+
+/**
+ * The name of this project. Used to exclude this repo as a lintable project.
+ */
+const THIS_PROJECT_NAME = 'module-lint';
+
+/**
+ * The root directory of this project.
+ */
+const THIS_PROJECT_DIRECTORY_PATH = path.resolve(__dirname, '..');
+
 /**
  * The number of milliseconds in an hour, used to determine when to pull the
  * latest changes for previously cached repositories.
  */
 export const ONE_HOUR = 60 * 60 * 1000;
+
+/**
+ * Wherever the tool was run.
+ */
+export const WORKING_DIRECTORY_PATH = process.cwd();
+
+/**
+ * The usage text printed when the user requests help or provides invalid input.
+ */
+export const USAGE_TEXT = `
+Analyzes one or more repos for divergence from a template repo.
+
+${THIS_PROJECT_NAME} OPTIONS [ARGUMENTS...]
+
+Pass the names of one or more MetaMask repositories to lint them, or pass
+nothing to lint all MetaMask repositories.
+`.trim();
+
+/**
+ * In order to lint a remote repository, that repository must be cloned first.
+ * This is the temporary directory where the clone lives.
+ */
+export const DEFAULT_CACHED_REPOSITORIES_DIRECTORY_PATH = path.join(
+  THIS_PROJECT_DIRECTORY_PATH,
+  'tmp/repositories',
+);
+
+/**
+ * The name of the template repository that project repositories will be
+ * compared to. The only such repository we have is the module template.
+ */
+export const DEFAULT_TEMPLATE_REPOSITORY_NAME = 'metamask-module-template';
+
+/**
+ * All of the remote MetaMask repositories that will be linted if a list is not
+ * explicitly provided.
+ */
+export const DEFAULT_PROJECT_NAMES = [
```

Comment:
Curious how this list was generated

Reply:
I generated this list from the code ownership spreadsheet. However, now that we have a new spreadsheet, perhaps I should use that instead? I wasn't sure what this should cover at the moment.

Reply:
Hmm. I guess we can figure it out later.

I was wondering because for a lot of these, we explicitly don't want to make them match the template because we're not really maintaining them (e.g. for the `ethjs-` packages, it's not really a goal to incorporate them fully into our set of packages, that would be a ton of work). And in some cases, we intentionally diverge from the template because the package isn't a TypeScript library (e.g. the actions).

Reply:
Maybe later on we could create an action template, and maintain a separate list for those

Reply:
List scaled back here: [b33c692](https://github.com/MetaMask/module-lint/pull/33/commits/b33c69208da0f243c3d945b625f81e27ecf10291)